Client/Frontend/Browser/TabManager.swift

```diff
@@ -251,12 +251,24 @@ class TabManager : NSObject {
 
 extension TabManager {
     class SavedTab: NSObject, NSCoding {
-        let url: NSURL?
         let isSelected: Bool
         let screenshot: UIImage?
+        var sessionData: SessionData?
 
         init?(browser: Browser, isSelected: Bool) {
-            self.url = browser.url
+            let currentItem = browser.webView?.backForwardList.currentItem
+            if self.sessionData == nil {
```

Comment:
This check isn't useful since `self.sessionData` will always be `nil` upon initialization. What I think you want to do is see if `browser.sessionData` is `nil`. If not, use that value instead of creating a new one here.